metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -49,6 +52,8 @@ final class BuildTargets(
     TrieMap.empty[BuildTargetIdentifier, util.Set[AbsolutePath]]
   private val inverseDependencySources =
     TrieMap.empty[AbsolutePath, Set[BuildTargetIdentifier]]
+  private val buildTargetGeneratedDirs: mutable.Set[AbsolutePath] =
+    mutable.Set.empty
```

Comment:
`mutable.Set` isn't thread-safe.
I don't if there is a thread-safe set collection for scala but it might be replaced at least by `TrieMap.empty[AbsolutePath, ()]`

Reply:
Maybe we should use `ConcurrentLinkedQueue` or something?
like: https://github.com/scalameta/metals/blob/924694f98d2c9774a436d4382b845185c2094293/metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala#L42

Reply:
fixed https://github.com/scalameta/metals/pull/3694/commits/a76f120e762fd4bab4a82ec861eb34f9af7502d4